pkg/util/cluster/cluster.go

```diff
@@ -202,6 +206,21 @@ func (c *Cluster) Create(ctx context.Context, vnetResourceGroup, clusterName str
 		"workerAddressPrefix":       {Value: workerSubnet},
 	}
 
+	// TODO: ick
```

Comment:
fundamental issue:
our E2E tests the RP in local development mode (i.e. running on CI), so it cannot by definition validate the gateway.

today we:
* run ARO RP binary on CI vm
* deploy cluster
* run cluster E2E

what's needed in the medium term:
* deploy *entire* ARO stack including gateway
* deploy cluster
* run cluster E2E

what's needed in the longer term:
* deploy *entire* ARO stack at current PROD version
* deploy cluster 1
* run cluster E2E
* deploy *entire* ARO stack at PR version
* upgrade cluster 1
* run cluster 1 E2E
* deploy cluster 2
* run cluster 2 E2E